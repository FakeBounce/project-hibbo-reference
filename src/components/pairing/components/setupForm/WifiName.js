import React from 'react';
import { string, func } from 'prop-types';

import * as Styles from './styles';

const WifiName = ({ label, action }) => {
  return (
    <Styles.SButtonControls
      activeOpacity={1}
      onPress={() => {
        action('wifiName', label);
      }}
    >
      <Styles.STextWifiName>{label}</Styles.STextWifiName>
    </Styles.SButtonControls>
  );
};

WifiName.propTypes = {
  label: string.isRequired,
  action: func.isRequired,
};

export default WifiName;
