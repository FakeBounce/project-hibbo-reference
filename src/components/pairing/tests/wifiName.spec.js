import React from 'react';
import renderer from 'react-test-renderer';

import WifiName from '../components/setupForm/WifiName';

describe('WifiName', () => {
  const fn = jest.fn();
  const component = renderer.create(
    <WifiName
      label=""
      action={fn}
    />,
  ).toJSON();

  it('should render the component', () => {
    expect(component).toMatchSnapshot();
  });
});
