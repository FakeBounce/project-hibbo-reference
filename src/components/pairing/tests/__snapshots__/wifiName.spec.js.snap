// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WifiName should render the component 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={undefined}
  >
    
  </Text>
</View>
`;
